examples/src/main/scala/fluence/codec/examples/PureCodecExample.scala

```diff
@@ -0,0 +1,138 @@
+      PureCodec.liftFuncEither((x: String) => Either.catchNonFatal(x.toInt).left.map(e => CodecError(e.getMessage))),
+      PureCodec.liftFuncEither((x: Int) => Either.catchNonFatal(x.toString).left.map(e => CodecError(e.getMessage)))
+    )
+
+
+    // For lawful strings – those which can be parsed into an integer the behavior hasn't really changed.
+    // Note that we receive Right(...) wrapped in the supplied monadic type.
+    {
+      val res: EitherT[Option, CodecError, Int] = str2intEitherCodec.direct[Option]("1024")
+      val resMonad = res.value
+      assert(res.toString == "EitherT(Some(Right(1024)))")
+      assert(resMonad.toString == "Some(Right(1024))")
+    }
+
+
+    // However, for strings that can't be parsed, we will receive Left(...) – which is a desired behavior!
+    {
+      val res: EitherT[Option, CodecError, Int] = str2intEitherCodec.direct[Option]("bar")
+      val resMonad = res.value
+      assert(res.toString == "EitherT(Some(Left(fluence.codec.CodecError: For input string: \"bar\")))")
+      assert(resMonad.toString == "Some(Left(fluence.codec.CodecError: For input string: \"bar\"))")
+    }
+
+
+    // It's also totally possible to perform an inverse transformation: after all, a codec is a bijection.
+    {
+      val res: EitherT[Id, CodecError, String] = str2intCodec.inverse[Id](720)
+      val resMonad: Id[Either[CodecError, String]] = res.value
+      assert(res.toString == "EitherT(Right(720))")
+      assert(resMonad.toString == "Right(720)")
+    }
+
+
+    // It's also possible to pass the to-be-converted value first, but perform the actual transformation only
+    // later on (using different enclosing monads if desired). To achieve this, `pointAt` method which returns a
+    // lazily evaluated function can be used.
+    {
+      val point: Point[Int] = str2intCodec.direct.pointAt("333")
+      val resId: EitherT[Id, CodecError, Int] = point[Id]()
+      val resOption: EitherT[Option, CodecError, Int] = point[Option]()
+      assert(resId.toString == "EitherT(Right(333))")
+      assert(resOption.toString == "EitherT(Some(Right(333)))")
+    }
+
+
+    // Sometimes, we might want to be able to compose two codecs together. Here we define an integer to boolean
+    // codec and compose it with one of the previously defined codecs. Yes, the int-to-bool codec is not really
+    // a bijection but we can put up with that for the sake of example.
+    val int2boolCodec: Bijection[Int, Boolean] = PureCodec.build[Int, Boolean](
+      (x: Int) => x != 0,
+      (x: Boolean) => if (x) 1 else 0
+    )
+    val str2boolCodec: Bijection[String, Boolean] = str2intCodec andThen int2boolCodec
+
+    {
+      val resA: EitherT[Id, CodecError, Boolean] = str2boolCodec.direct[Id]("100")
+      val resB = str2boolCodec.inverse[Option](true)
+      assert(resA.toString == "EitherT(Right(true))")
+      assert(resB.toString == "EitherT(Some(Right(1)))")
+    }
+
+
+    // TODO: describe `runF` and `toKleisli`
```

Comment:
Actually, it seems the `runF` example quoted on the readme page is off. When I've tried to use `runF` accordingly to it, the compilation error has occurred. The code was:
```
val x = str2intCodec.direct.runF[Id]("foo")
println(x)
```

The error was:
```
[error] /Users/xdralex/Work/fluence/codec/examples/src/main/scala/fluence/codec/examples/PureCodecExample.scala:55:41: could not find implicit value for parameter F: cats.MonadError[cats.Id, _ >: fluence.codec.CodecError]
[error]     val x = str2intCodec.direct.runF[Id]("foo")
[error]                                         ^
[error] one error found
```

Not sure how the correct usage pattern should look like.